homeassistant/components/renault/config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Config flow to configure Renault component."""
+from __future__ import annotations
+
+from typing import Any
+
+from renault_api.const import AVAILABLE_LOCALES
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_KAMEREON_ACCOUNT_ID, CONF_LOCALE, DOMAIN
+from .renault_hub import RenaultHub
+
+
+class RenaultFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Renault config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the Renault config flow."""
+        self.renault_config: dict[str, Any] = {}
+        self.renault_hub: RenaultHub | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a Renault config flow start.
+
+        Ask the user for API keys.
+        """
+        if user_input:
+            locale = user_input[CONF_LOCALE]
+            self.renault_config.update(user_input)
+            self.renault_config.update(AVAILABLE_LOCALES[locale])
+            self.renault_hub = RenaultHub(self.hass, locale)
+            if not await self.renault_hub.attempt_login(
+                user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+            ):
+                return self._show_user_form({"base": "invalid_credentials"})
+            return await self.async_step_kamereon()
+        return self._show_user_form()
+
+    def _show_user_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
+        """Show the API keys form."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_LOCALE): vol.In(AVAILABLE_LOCALES.keys()),
```

Comment:
it defaults compares `in` checks against the keys

```suggestion
                    vol.Required(CONF_LOCALE): vol.In(AVAILABLE_LOCALES),
```

Reply:
And if you make it a dictionary, the frontend will show it as `key => value`

Reply:
Ah I see now that it's used as a dictionary so `keys()` is correct